### PR TITLE
Profileinfopopulate

### PR DESCRIPTION
**Pull request by melinamboedecker**

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -113,9 +113,9 @@ function App() {
             <Route exact path="/addpost">
               <AddPost />
             </Route>
-            <Route exact path="/profile">
+            {/* <Route exact path="/profile">
```
Comment (vasylynash) [inline client/src/App.js]: Duplicate? Remove

File: client/src/components/post/FullPost.js [hunk 2]
```diff
@@ -114,6 +114,8 @@ const  FullPost= () => {
     });
     const post = data?.post || {};
 
+    console.log('post ->     ', post)
```
Comment (vasylynash) [inline client/src/components/post/FullPost.js]: Remove console.log

File: client/src/components/profile/Index.js [hunk 3]
```diff
@@ -2,8 +2,23 @@ import React from 'react';
 import Info from './Info';
 import Security from './Security';
 import UserPosts from './UserPosts';
+import Auth from '../../utils/auth';
+import { useQuery } from '@apollo/client';
+import { QUERY_SINGLE_USER } from '../../utils/queries';
 
 function Index(props) {
+    console.log('index')
```
Comment (vasylynash) [inline client/src/components/profile/Index.js]: Remove console.log

File: client/src/components/profile/Index.js [hunk 4]
```diff
@@ -2,8 +2,23 @@ import React from 'react';
...
+
+    const user = Auth.getProfile().data;
+    console.log('user id ->   ', user._id);
+
+    const { loading, err, data } = useQuery(QUERY_SINGLE_USER, {
+        variables: { id: user._id },
+        skip: true
+        // variables: { id: '61400ff61f3834d48cc966d0' }
```
Comment (vasylynash) [inline client/src/components/profile/Index.js]: Remove

File: client/src/components/profile/Index.js [hunk 5]
```diff
@@ -2,8 +2,23 @@ import React from 'react';
...
+    })
+    console.log('data ->   ', data)
```
Comment (vasylynash) [inline client/src/components/profile/Index.js]: Remove console.log

File: client/src/components/profile/Security.js [hunk 6]
```diff
@@ -2,8 +2,16 @@ import React, { useState } from 'react';
...
 import { Line } from '../posts/SearchPost';
+import {  useMutation } from '@apollo/client';
+import { CHANGE_PASSWORD } from '../../utils/mutations';
+import Auth from '../../utils/auth';
 
 function Security() {
+    console.log('security')
+
+    const user = Auth.getProfile().data
+    console.log('user ->     ', user)
```
Comment (vasylynash) [inline client/src/components/profile/Security.js]: Remove console.log

File: client/src/components/profile/UserPosts.js [hunk 7]
```diff
@@ -5,7 +5,7 @@ import { Link } from 'react-router-dom';
 import UserPost from '../posts/UserPost';
 
 function UserPosts() {
-
+    console.log('userposts')
```
Comment (vasylynash) [inline client/src/components/profile/UserPosts.js]: REmove console.log

File: client/src/pages/Home.js [hunk 8]
```diff
@@ -6,6 +6,8 @@ import SearchProvider from '../utils/SearchContext';
 import { VerticalDiv } from './Landing';
 
 const Home = () => {
+    
+    console.log('HOME')
```
Comment (vasylynash) [inline client/src/pages/Home.js]: Remove console.log

File: client/src/pages/Information.js [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+
+const FormContainer = styled.div`
+    .textfield {
+        margin-top: 10px;
+    }
+`
+
+const Information = () => {
+    console.log('info')
+    // console.log('currentuser ->   ', currentUser)
```
Comment (vasylynash) [inline client/src/pages/Information.js]: REmove

File: client/src/pages/Information.js [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+        firstName: '',
+        lastName: '',
+        email: '',
+        className: '', 
+        bio: '',
+        linkedIn: '',
+        gitHub: '',
+    });
+
+    // setFormState({
```
Comment (vasylynash) [inline client/src/pages/Information.js]: Remove

File: client/src/pages/Profile.js [hunk 11]
```diff
@@ -46,6 +47,7 @@ const ProfilePic = styled.div`
 `
 
 const Profile = () => {
+    console.log('profile')
```
Comment (vasylynash) [inline client/src/pages/Profile.js]: Remove

File: client/src/pages/Profile.js [hunk 12]
```diff
@@ -80,6 +82,7 @@ const Profile = () => {
     return ( 
         <>
         <GlobalStyle/>
+        <UserProvider>
```
Comment (vasylynash) [inline client/src/pages/Profile.js:84]: Are you using this?

File: client/src/utils/UserContext.js [hunk 13]
```diff
@@ -0,0 +1,42 @@
+import React, { useState, useContext } from 'react';
```
Comment (vasylynash) [inline client/src/utils/UserContext.js:1]: Are you using UserProvider?

**Review by vasylynash — CHANGES_REQUESTED** [review]

File: client/src/utils/UserContext.js (hunk 13 above)
Comment (melinamboedecker) [reply client/src/utils/UserContext.js:1]: breaks if remove

File: client/src/pages/Profile.js (hunk 12 above)
Comment (melinamboedecker) [reply client/src/pages/Profile.js:84]: doesn't work if remove this tag

File: client/src/pages/Profile.js (hunk 11 above)
Comment (melinamboedecker) [reply client/src/pages/Profile.js]: removed

File: client/src/pages/Information.js (hunk 10 above)
Comment (melinamboedecker) [reply client/src/pages/Information.js]: removed file , wasn't using it

File: client/src/pages/Information.js (hunk 9 above)
Comment (melinamboedecker) [reply client/src/pages/Information.js]: removed file

File: client/src/components/profile/Index.js (hunk 5 above)
Comment (melinamboedecker) [reply client/src/components/profile/Index.js]: removed

File: client/src/components/profile/Index.js (hunk 4 above)
Comment (melinamboedecker) [reply client/src/components/profile/Index.js]: removed

File: client/src/components/profile/Index.js (hunk 3 above)
Comment (melinamboedecker) [reply client/src/components/profile/Index.js]: removed

File: client/src/components/post/FullPost.js (hunk 2 above)
Comment (melinamboedecker) [reply client/src/components/post/FullPost.js]: removed

File: client/src/App.js (hunk 1 above)
Comment (melinamboedecker) [reply client/src/App.js]: removed

**Review by vasylynash — APPROVED** [review]

**Review by Dimavich — APPROVED** [review]

**Comment by melinamboedecker** [comment]
[place for users to see current profile info or this can be pre-populated in the info page form fields.](https://trello.com/c/76kKFNuG/35-place-for-users-to-see-current-profile-info-or-this-can-be-pre-populated-in-the-info-page-form-fields)

**Pull request merged**